R/pool.R

```diff
@@ -45,118 +53,178 @@ Pool <- R6Class("Pool",
       }
       reg.finalizer(self,
         function(self) {
-          protectDefaultScheduler({
-            freeEnv <- private$freeObjects
-            if (length(freeEnv) > 0) {
-              self$close()
-            }
-          })
+          if (self$valid) {
+            warning("Closing leaked pool.")
+            self$close()
+          }
         },
         onexit = TRUE)
```

Comment:
update for future reference: keep `onexit = TRUE`. This only has a negative effect on the app's performance if you have already so many leaked objects that cleanup takes a while. But if you're in that state, you're already in trouble. On the other hand, `onexit = TRUE` guarantees that the external state is clean and ready to go.